lib/core/src/lib/viewer/components/viewer.component.spec.ts

```diff
@@ -695,4 +696,67 @@ describe('ViewerComponent', () => {
             expect(component.downloadFile.emit).toHaveBeenCalled();
         }));
     });
+
+    describe('ViewerComponent toolbar separators', () => {
+        let component: ViewerComponent<any>;
+        let fixture: ComponentFixture<ViewerComponent<any>>;
+
+        beforeEach(async () => {
+            await TestBed.configureTestingModule({
+                imports: [MatButtonModule, MatIconModule, MatDialogModule],
+                declarations: [ViewerComponent]
+            }).compileComponents();
+        });
+
+        beforeEach(() => {
+            fixture = TestBed.createComponent(ViewerComponent);
+            component = fixture.componentInstance;
+            component.showViewer = true;
+            fixture.detectChanges();
+        });
+
+        const createToolbarTest = (
+            allowRightSidebar: boolean,
+            hideInfoButton: boolean,
+            allowFullScreen: boolean,
+            allowGoBack: boolean,
+            expectedSeparatorCount: number
+        ) => {
+            it(`should have correct UI state when [R:${allowRightSidebar}, I:${hideInfoButton}, F:${allowFullScreen}, B:${allowGoBack}]`, () => {
+                component.allowRightSidebar = allowRightSidebar;
+                component.hideInfoButton = hideInfoButton;
+                component.allowFullScreen = allowFullScreen;
+                component.allowGoBack = allowGoBack;
```

Comment:
I think this button can be rendered before the divider so I think we should include it in the tests
```suggestion
                component.allowGoBack = allowGoBack; mnuOpenWith
                component.mnuOpenWith = ...
```